it/controllers/benefits/reimbursed/OtherBenefitsAmountControllerISpec.scala

```diff
@@ -0,0 +1,499 @@
+          errorAboveElementCheck(user.specificExpectedResults.get.emptyErrorText)
+        }
+
+        "should render the amount page with invalid format text when input is in incorrect format" which {
+          implicit lazy val result: WSResponse = {
+            authoriseAgentOrIndividual(user.isAgent)
+            dropEmploymentDB()
+            insertCyaData(employmentUserData(isPrior = true, cyaModel(Some(benefits(fullReimbursedCostsVouchersAndNonCashModel)))), userRequest)
+            urlPost(pageUrl(taxYearEOY), welsh = user.isWelsh, headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYear)), body = Map("amount" -> "abc"))
+          }
+
+          implicit def document: () => Document = () => Jsoup.parse(result.body)
+
+          "has an BAD_REQUEST status" in {
+            result.status shouldBe BAD_REQUEST
+          }
+
+          titleCheck(user.specificExpectedResults.get.expectedErrorTitle)
+          h1Check(user.specificExpectedResults.get.expectedHeading)
+          captionCheck(expectedCaption)
+          textOnPageCheck(previousExpectedContent, previousAmountTextSelector)
+          textOnPageCheck(amountHint, hintTextSelector)
+          textOnPageCheck(poundPrefixText, poundPrefixSelector)
+          inputFieldValueCheck("abc", inputSelector)
+
+          buttonCheck(user.commonExpectedResults.continue, continueButtonSelector)
+          formPostLinkCheck(continueLink, continueButtonFormSelector)
+          welshToggleCheck(user.isWelsh)
+
+          errorSummaryCheck(user.specificExpectedResults.get.invalidFormatErrorText, expectedErrorHref)
+          errorAboveElementCheck(user.specificExpectedResults.get.invalidFormatErrorText)
+        }
+
+        "should render the amount page with max error when input > 99,999,999,999" which {
+          implicit lazy val result: WSResponse = {
+            authoriseAgentOrIndividual(user.isAgent)
+            dropEmploymentDB()
+            insertCyaData(employmentUserData(isPrior = true, cyaModel(Some(benefits(fullReimbursedCostsVouchersAndNonCashModel)))), userRequest)
+            urlPost(pageUrl(taxYearEOY), welsh = user.isWelsh, headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYear)),
+              body = Map("amount" -> "100,000,000,000"))
+          }
+
+          implicit def document: () => Document = () => Jsoup.parse(result.body)
+
+          "has an BAD_REQUEST status" in {
+            result.status shouldBe BAD_REQUEST
+          }
+
+          titleCheck(user.specificExpectedResults.get.expectedErrorTitle)
+          h1Check(user.specificExpectedResults.get.expectedHeading)
+          captionCheck(expectedCaption)
+          textOnPageCheck(previousExpectedContent, previousAmountTextSelector)
+          textOnPageCheck(amountHint, hintTextSelector)
+          textOnPageCheck(poundPrefixText, poundPrefixSelector)
+          inputFieldValueCheck("100,000,000,000", inputSelector)
```

Comment:
same here

Reply:
as discussed not needed as the inputFieldValueCheck would fail if its not there